cli/azd/pkg/installer/installed_by.go

```diff
@@ -0,0 +1,66 @@
+package installer
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+const cInstalledByFileName = ".installed-by.txt"
+
+type InstallType string
+
+const InstallTypeUnknown InstallType = ""
+const InstallTypePs InstallType = "install-azd.ps1"
+const InstallTypeSh InstallType = "install-azd.sh"
+const InstallTypeBrew InstallType = "brew"
+const InstallTypeChoco InstallType = "choco"
+const InstallTypeWinget InstallType = "winget"
+const InstallTypeDeb InstallType = "deb"
+const InstallTypeRpm InstallType = "rpm"
+
+func InstalledBy() InstallType {
+	raw := RawInstalledBy()
+
+	switch raw {
+	case string(InstallTypePs):
+		return InstallTypePs
+	case string(InstallTypeSh):
+		return InstallTypeSh
+	case string(InstallTypeBrew):
+		return InstallTypeBrew
+	case string(InstallTypeChoco):
+		return InstallTypeChoco
+	case string(InstallTypeWinget):
+		return InstallTypeWinget
+	case string(InstallTypeDeb):
+		return InstallTypeDeb
+	case string(InstallTypeRpm):
+		return InstallTypeRpm
+	default:
+		return InstallTypeUnknown
+	}
+}
+
+func RawInstalledBy() string {
```

Comment:
Let's add some doc comments to these exported methods.

They can be simple like:

// RawInstalledBy returns the contents of the installed by text file, after removing leading and trailing whitespace.